conmon/conmon.c

```diff
@@ -1397,6 +1408,25 @@ int main(int argc, char *argv[])
 			nexitf("Failed to write %s to exit file: %s", status_str, err->message);
 	}
 
+	if (opt_exit_namespace_args) {
+		for (fd = 0; opt_exit_namespace_args[fd]; fd++) {
+			int type, ns;
+			char *endptr = NULL;
+
+			errno = 0;
+			ns = strtol(opt_exit_namespace_args[fd], &endptr, 10);
+			if (errno != 0 || *endptr != '\0')
+				nexitf("unable to parse %s", opt_exit_namespace_args[fd]);
+
+			type = ioctl(ns, NS_GET_NSTYPE);
+			if (unshare(type) < 0)
```

Comment:
Why are we calling unshare if we want to join the namespace? Shouldn't just setns be enough? In either case, I think adding some comments here will help :)

Reply:
Should this just be restricted to the mount namespace to keep the scope narrow for the specific problem we are trying to fix?

Reply:
without the unshare I get `EINVAL` from the setns.  This is the same pattern we are using in Podman.

I can restrict it to the mount namespace, although I preferred to make it more generic in case we will need in future to join some different namespace.